leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/extensions/MessageDelayerExtension.java

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Achim Kraus (Bosch Software Innovations GmbH) - Initial contribution
+ ******************************************************************************/
+package org.eclipse.leshan.server.demo.extensions;
+
+import java.util.Random;
+
+import org.eclipse.californium.core.coap.EmptyMessage;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.network.Endpoint;
+import org.eclipse.californium.core.network.interceptors.MessageInterceptor;
+import org.eclipse.leshan.server.californium.impl.LeshanServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Message delayer extensions. Delays processing of messages to simulate processing under stress condition. Used for
+ * robustness test of clients. The extension toggles between delayed and undelayed processing to see, if the client is
+ * able to work under "normal" condition when it was "stressed before.
+ */
+public class MessageDelayerExtension implements MessageInterceptor, LeshanServerExtension {
```

Comment:
Because:
- not everyone use linux
- not to read the documentation of an other tool

I'm not saying, that all extension currently contained in the PR are "usefull" for all.
But the principle of extension may be usefull. Therefore I proposed to shrink it, if it would then have chance.

(The "noise test stuff" will only be usefull together with the ToggleExtension. This temporary enables/disables the noise so you can check, if a device get connected again after the noise.)